### PR TITLE
reenable duedate field

### LINKED ISSUE #557: Due date for task
### Describe the feature

Re-Enable the due date logic and sync the field with the API.

### Why is this feature needed?

A user needs to set a due date for a task

### Additional context

[The logic was disabled by this pr.](https://github.com/helpwave/web/pull/520)

### PR DESCRIPTION
**Pull request by DasProffi**
## Which issues does this pull request close?
closes #557

### CODE REVIEW

**Comment by DasProffi** [comment]
Waiting for a decision on how duedates are removed in the backend interface

**Comment by MaxSchaefer** [comment]
> Waiting for a decision on how duedates are removed in the backend interface

Please open this PR. The removal of due dates will be a future enhancement.

File: tasks/utils/timeConversion.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  const milliseconds = seconds * 1000 + nanos / 1e6
+
+  return new Date(milliseconds)
+}
+
+export const timestampToDateOptional = (timestamp?: Timestamp): Date | undefined => {
+  if (timestamp) {
+    return timestampToDate(timestamp)
+  }
```
Comment (use-to) [inline tasks/utils/timeConversion.ts:15]:
```suggestion
  }
  
```

File: tasks/utils/timeConversion.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  return undefined
+}
+
+export const dateToTimestamp = (date: Date): Timestamp => {
+  const timestamp = new Timestamp()
+  timestamp.setSeconds(Math.floor(date.getTime() / 1000)) // Convert to seconds
+  timestamp.setNanos(date.getMilliseconds() * 1e6) // Convert milliseconds to nanoseconds
+  return timestamp
```
Comment (use-to) [inline tasks/utils/timeConversion.ts:23]:
```suggestion
  const timestamp = new Timestamp()
  
  timestamp.setSeconds(Math.floor(date.getTime() / 1000)) // Convert to seconds
  timestamp.setNanos(date.getMilliseconds() * 1e6) // Convert milliseconds to nanoseconds
  
  return timestamp
```

File: tasks/utils/timeConversion.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+export const dateToTimestampOptional = (date?: Date): Timestamp | undefined => {
+  if (date) {
+    return dateToTimestamp(date)
+  }
```
Comment (use-to) [inline tasks/utils/timeConversion.ts:29]:
```suggestion
  }
  
```

**Review by use-to — APPROVED** [review]

**Review by use-to — APPROVED** [review]
For initial creation setting the status to in progress or done changes nothing on creation. We need to grey that selection on creation or move it directly after it or even better on create.

**Comment by DasProffi** [comment]
> For initial creation setting the status to in progress or done changes nothing on creation. We need to grey that selection on creation or move it directly after it or even better on create.
> 
> @DasProffi it's up to you whether you want to fix it in this pull request or not. Please create and link a proper issue if you decide not to implement this in here.

Moved to seperate issue see #623

**Pull request merged**